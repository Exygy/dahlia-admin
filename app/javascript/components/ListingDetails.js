import React from 'react'
import ListingDetailsContentCard from './ListingDetailsContentCard'
import ListingDetailsContentTable from './ListingDetailsContentTable'

const ListingDetails = ({ listing }) => {
  // to do: move fields into own file to import and map
  let detailsFields = [
    'Owner.Name',
    'Name',
    'Account.Name',
    'Application_Due_Date',
    'Number_of_Applications',
    'Status',
    'Building__r.Name',
    'Min_Occupancy',
    'Max_Occupancy',
    'Min_BR',
    'Max_BR',
    'Lottery_Winners',
    'Lottery_Results',
    'Min_Income',
    'Max_Income',
  ]

  let buildingInformationFields = [
    'Building_Name',
    'Project_ID',
    'Building_Street_Address',
    'Building_City',
    'Building_State',
    'Building_Zip_Code',
    'Neighborhood',
    'Developer',
    'Building_URL',
    'Year_Built',
    'Description',
    'Lottery_Preferences'
  ]

  let lotteryPreferencesFields = [
    'Lottery_Preference.Name',
    'Description',
    'PDF_URL',
    'Order',
    'Available_Units'
  ]

  let aafFields = [
    'Accessibility',
    'Fee',
    'Amenities',
    'Deposit_Min',
    'Costs_Not_Included',
    'Deposit_Max'
  ]

  let lotteryInfoFields = [
    'Lottery_Date',
    'Lottery_Results_Date',
    'Lottery_Venue',
    'Lottery_Status',
    'Lottery_Street_Address',
    'Lottery_Summary',
    'Lottery_City',
    'Lottery_URL',
  ]

  let appInfoFields = [
    'Application_Phone',
    'Office_Hours',
    'Application_Organization',
    'Office_Closed',
    'Application_Street_Address',
    'Application_City',
    'Application_State',
    'Application_Postal_Code',
    'Organization_URL',
    'Download_URL'
  ]

  let agentDevInfoFields= [
    'Leasing_Agent_Name',
    'Leasing_Agent_Title',
    'Leasing_Agent_Email',
    'Leasing_Agent_Phone',
    'Preference_Detail',
    'Rental_Assistance',
  ]

  let eligibilityRulesFields = [
    'Building_Selection_Criteria',
    'Eviction_History',
    'Criminal_History',
    'Credit_Rating'
  ]

  let additionalInfoFields = [
    'Required_Documents',
    'Smoking_Policy',
    'Legal_Disclaimers',
    'Pet_Policy'
  ]

  let openHousesFields = [
    'Date',
    'Start_Time',
    'End_Time'
  ]

  let infoSessionsFields = [
    'Date',
    'Start_Time',
    'End_Time',
    'Venue',
    'Street_Address',
    'City'
  ]

  return (
    <div>
      <ListingDetailsContentCard
        listing={listing}
        title='Details'
        fields={detailsFields}
      />
      <ListingDetailsContentCard
        listing={listing}
        title='Building Information'
        fields={buildingInformationFields}
      />
      {(() => {
        if (listing.Listing_Lottery_Preferences) {
          return (
            <ListingDetailsContentTable
              listing={listing}
              title='Listing Preferences'
              table='Listing_Lottery_Preferences'
              fields={lotteryPreferencesFields}
            />
          )
        }
      })()}
      <ListingDetailsContentCard
        listing={listing}
        title='Accessibility, Amenities, Fees'
        fields={aafFields}
      />
      <ListingDetailsContentCard
        listing={listing}
        title='Lottery Information'
        fields={lotteryInfoFields}
      />
      <ListingDetailsContentCard
        listing={listing}
        title='Application Information'
        fields={appInfoFields}
      />
      <ListingDetailsContentCard
        listing={listing}
        title='Leasing Agent and Developer Information'
        fields={agentDevInfoFields}
      />
      <ListingDetailsContentCard
        listing={listing}
        title='Additional Eligibility Rules'
        fields={eligibilityRulesFields}
      />
      <ListingDetailsContentCard
        listing={listing}
        title='Additional Information'
        fields={additionalInfoFields}
      />
      {(() => {
        if (listing.Open_Houses) {
          return (
            <ListingDetailsContentTable
              listing={listing}
              title='Open Houses'
              table='Open_Houses'
              fields={openHousesFields}
            />
          )
        }
      })()}
      {(() => {
        if (listing.Information_Sessions) {
          return (
            <ListingDetailsContentTable
              listing={listing}
              title='Information Sessions'
              table='Information_Sessions'
              fields={infoSessionsFields}
            />
          )
        }
      })()}
    </div>
  )
}

export default ListingDetails
