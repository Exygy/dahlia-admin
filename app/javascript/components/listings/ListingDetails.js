import React from 'react'
import { isEmpty } from 'lodash'
import ListingDetailsContentCard from './ListingDetailsContentCard'
import ListingDetailsContentTable from './ListingDetailsContentTable'

import {
  detailsFields,
  buildingInformationFields,
  preferencesFields,
  aafFields,
  appInfoFields,
  agentDevInfoFields,
  eligibilityRulesFields,
  additionalInfoFields,
  openHousesFields,
  infoSessionsFields
} from './fields'

const ListingDetails = ({ listing }) => {
  const Card = (cardProps) => <ListingDetailsContentCard listing={listing} {...cardProps} />
  const Table = (tableProps) => <ListingDetailsContentTable listing={listing} {...tableProps} />

  return (
    <div>
      <Card title='Details' fields={detailsFields} />
      <Card title='Building Information' fields={buildingInformationFields} />
      { !isEmpty(listing.preferences) && (
        <Table title='Listing Preferences'
          table='listing_preferences'
          fields={preferencesFields} />
      )
      }
      <Card title='Accessibility, Amenities, Fees' fields={aafFields} />
      <Card title='Application Information' fields={appInfoFields} />
      <Card title='Leasing Agent and Developer Information' fields={agentDevInfoFields} />
      <Card title='Additional Eligibility Rules' fields={eligibilityRulesFields} />
      <Card title='Additional Information' fields={additionalInfoFields} />
      { !isEmpty(listing.open_houses) && (
        <Table title='Open Houses'
          table='open_houses'
          fields={openHousesFields} />
      )
      }
      { !isEmpty(listing.information_sessions) && (
        <Table title='Information Sessions'
          table='information_sessions'
          fields={infoSessionsFields} />
      )
      }
    </div>
  )
}

export default ListingDetails
