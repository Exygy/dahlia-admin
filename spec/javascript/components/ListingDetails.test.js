import React from 'react'
import renderer from 'react-test-renderer'
import ListingDetails from 'components/ListingDetails'

test('Should render ListingDetails', () => {
  /* More data is required here */
  const listings = {
    Listing_Lottery_Preferences: [],
    Open_Houses: true,
    Information_Sessions: [],
    'Owner.Name': 'xxx'
  }
  const component = renderer.create(
    <ListingDetails listing = { listings }/ >,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
