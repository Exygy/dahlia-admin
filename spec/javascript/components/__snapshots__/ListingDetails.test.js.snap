// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ListingDetails 1`] = `
<div>
  <div
    className="content-card padding-bottom-none margin-bottom--half bg-trans"
  >
    <h4
      className="content-card_title t-serif"
    >
      Details
    </h4>
    <ul
      className="content-grid"
    >
      <li
        className="content-item"
      >
        <div
          className="content-card"
        >
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Owner
            </h4>
            <p>
              xxx
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Application Due Date
            </h4>
            <p>
              04/20/2018
            </p>
          </div>
        </div>
      </li>
      <li
        className="content-item"
      >
        <div
          className="content-card"
        />
      </li>
    </ul>
  </div>
  <div
    className="content-card padding-bottom-none margin-bottom--half bg-trans"
  >
    <h4
      className="content-card_title t-serif"
    >
      Building Information
    </h4>
    <ul
      className="content-grid"
    >
      <li
        className="content-item"
      >
        <div
          className="content-card"
        />
      </li>
      <li
        className="content-item"
      >
        <div
          className="content-card"
        />
      </li>
    </ul>
  </div>
  <div
    className="content-card"
  >
    <h4
      className="content-card_title t-serif"
    >
      Listing Preferences
    </h4>
    <table>
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Description
          </th>
          <th>
            PDF URL
          </th>
          <th>
            Order
          </th>
          <th>
            Available Units
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
  <div
    className="content-card padding-bottom-none margin-bottom--half bg-trans"
  >
    <h4
      className="content-card_title t-serif"
    >
      Accessibility, Amenities, Fees
    </h4>
    <ul
      className="content-grid"
    >
      <li
        className="content-item"
      >
        <div
          className="content-card"
        />
      </li>
      <li
        className="content-item"
      >
        <div
          className="content-card"
        />
      </li>
    </ul>
  </div>
  <div
    className="content-card padding-bottom-none margin-bottom--half bg-trans"
  >
    <h4
      className="content-card_title t-serif"
    >
      Lottery Information
    </h4>
    <ul
      className="content-grid"
    >
      <li
        className="content-item"
      >
        <div
          className="content-card"
        >
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Lottery Date
            </h4>
            <p>
              04/20/2018
            </p>
          </div>
          <div
            className="margin-bottom--half"
          >
            <h4
              className="t-sans t-small t-bold no-margin"
            >
              Lottery Results Date
            </h4>
            <p>
              04/20/2018
            </p>
          </div>
        </div>
      </li>
      <li
        className="content-item"
      >
        <div
          className="content-card"
        />
      </li>
    </ul>
  </div>
  <div
    className="content-card padding-bottom-none margin-bottom--half bg-trans"
  >
    <h4
      className="content-card_title t-serif"
    >
      Application Information
    </h4>
    <ul
      className="content-grid"
    >
      <li
        className="content-item"
      >
        <div
          className="content-card"
        />
      </li>
      <li
        className="content-item"
      >
        <div
          className="content-card"
        />
      </li>
    </ul>
  </div>
  <div
    className="content-card padding-bottom-none margin-bottom--half bg-trans"
  >
    <h4
      className="content-card_title t-serif"
    >
      Leasing Agent and Developer Information
    </h4>
    <ul
      className="content-grid"
    >
      <li
        className="content-item"
      >
        <div
          className="content-card"
        />
      </li>
      <li
        className="content-item"
      >
        <div
          className="content-card"
        />
      </li>
    </ul>
  </div>
  <div
    className="content-card padding-bottom-none margin-bottom--half bg-trans"
  >
    <h4
      className="content-card_title t-serif"
    >
      Additional Eligibility Rules
    </h4>
    <ul
      className="content-grid"
    >
      <li
        className="content-item"
      >
        <div
          className="content-card"
        />
      </li>
      <li
        className="content-item"
      >
        <div
          className="content-card"
        />
      </li>
    </ul>
  </div>
  <div
    className="content-card padding-bottom-none margin-bottom--half bg-trans"
  >
    <h4
      className="content-card_title t-serif"
    >
      Additional Information
    </h4>
    <ul
      className="content-grid"
    >
      <li
        className="content-item"
      >
        <div
          className="content-card"
        />
      </li>
      <li
        className="content-item"
      >
        <div
          className="content-card"
        />
      </li>
    </ul>
  </div>
  <div
    className="content-card"
  >
    <h4
      className="content-card_title t-serif"
    >
      Open Houses
    </h4>
    <table>
      <thead>
        <tr>
          <th>
            Date
          </th>
          <th>
            Start Time
          </th>
          <th>
            End Time
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
  <div
    className="content-card"
  >
    <h4
      className="content-card_title t-serif"
    >
      Information Sessions
    </h4>
    <table>
      <thead>
        <tr>
          <th>
            Date
          </th>
          <th>
            Start Time
          </th>
          <th>
            End Time
          </th>
          <th>
            Venue
          </th>
          <th>
            Street Address
          </th>
          <th>
            City
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
